aspnetcore/grpc/performance.md

```diff
@@ -156,6 +156,28 @@ The preceding code configures a channel that sends a keep alive ping to the serv
 
 :::moniker-end
 
+## Flow control
+
+HTTP/2 flow control is a feature that prevents apps from being overwhelmed with data. When using flow control:
+
+* Each app has an available buffer window. The buffer window is how much data the app can receive at once.
+* Flow control activates if the buffer window is filled up. When activated, the sending app pauses sending more data.
+* Once the receiving app has processed data then space in the buffer window is available. The sending app resumes sending data.
+
+Flow control can have a negative impact on performance when sending and receiving large messages. If the buffer window isn't large enough, or there is latency between the client and server, then data can be sent in start/stop bursts. This can be fixed by increasing HTTP/2 flow control window size. In Kestrel this is configured with <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Http2Limits.InitialConnectionWindowSize> and <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Http2Limits.InitialStreamWindowSize> at app startup.
```

Comment:
```suggestion
Flow control can have a negative impact on performance when sending and receiving large messages. If the buffer window isn't large enough or there's latency between the client and server, then data can be sent in start/stop bursts. This can be fixed by increasing the HTTP/2 flow control window size. In Kestrel, this is configured with <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Http2Limits.InitialConnectionWindowSize> and <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Http2Limits.InitialStreamWindowSize> at app startup.
```